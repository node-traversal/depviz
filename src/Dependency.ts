export default class Dependency {
    private readonly _name: string;
    private readonly _deps: Dependency[];

    constructor(name: string, deps: Dependency[]) {
        this._name = name;
        this._deps = deps;
    }

    public get name(): string {
        return this._name;
    }

    public get deps(): Dependency[] {
        return this._deps;
    }
}