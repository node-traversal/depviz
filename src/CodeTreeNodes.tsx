import * as React from "react";
import TreeItem from "@material-ui/lab/TreeItem";
import Dependency from "./Dependency";

type CodeTreeNodesProps = {
    nodes: Dependency[];
    path: string;
}

const CodeTreeNodes: React.FC<CodeTreeNodesProps> = ({nodes, path}) => {
    const elements: React.ReactElement[] = nodes.map((node: Dependency): React.ReactElement => {
        const key = path + '.' + node.name;
        const label = node.name;
        let nodes: React.ReactElement;

        if (node.deps.length > 0) {
            nodes = (
                <TreeItem key={key} nodeId={key} label={label}>
                    <CodeTreeNodes nodes={node.deps} path={key} />
                </TreeItem>
            );
        } else {
            nodes = <TreeItem key={key} nodeId={key} label={label} />;
        }

        return nodes;
    });

    return <React.Fragment>{elements}</React.Fragment>;
};

export default CodeTreeNodes;
