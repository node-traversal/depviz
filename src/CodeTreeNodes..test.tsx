import React from 'react';
import CodeTreeNodes from './CodeTreeNodes';
import { shallow } from 'enzyme';
import Dependency from "./Dependency";

it('renders empty code-tree nodes', () => {
    const wrapper = shallow(<CodeTreeNodes nodes={[]} path=""/>);
    expect(wrapper.debug()).toMatchSnapshot();
});

it('renders nested code-tree nodes', () => {
    const dependencies = [
        new Dependency('the-node-name', [
            new Dependency('down-stream-node', [])
        ])
    ];

    const wrapper = shallow(<CodeTreeNodes nodes={dependencies} path=""/>);
    expect(wrapper.debug()).toMatchSnapshot();
});

it('renders leaf code-tree nodes', () => {
    const dependencies = [
        new Dependency('the-node-name', [])
    ];

    const wrapper = shallow(<CodeTreeNodes nodes={dependencies} path=""/>);
    expect(wrapper.debug()).toMatchSnapshot();
});

it('renders terminal leaf code-tree nodes', () => {
    const dependencies = [
        new Dependency('the-node-name', [])
    ];

    const wrapper = shallow(<CodeTreeNodes nodes={dependencies} path="@the-package"/>);
    expect(wrapper.debug()).toMatchSnapshot();
});
