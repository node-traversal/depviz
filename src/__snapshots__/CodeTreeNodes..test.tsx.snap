// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty code-tree nodes 1`] = `"<Fragment />"`;

exports[`renders leaf code-tree nodes 1`] = `
"<Fragment>
  <WithStyles(ForwardRef(TreeItem)) nodeId=\\".the-node-name\\" label=\\"the-node-name\\" />
</Fragment>"
`;

exports[`renders nested code-tree nodes 1`] = `
"<Fragment>
  <WithStyles(ForwardRef(TreeItem)) nodeId=\\".the-node-name\\" label=\\"the-node-name\\">
    <CodeTreeNodes nodes={{...}} path=\\".the-node-name\\" />
  </WithStyles(ForwardRef(TreeItem))>
</Fragment>"
`;

exports[`renders terminal leaf code-tree nodes 1`] = `
"<Fragment>
  <WithStyles(ForwardRef(TreeItem)) nodeId=\\"@the-package.the-node-name\\" label=\\"the-node-name\\" />
</Fragment>"
`;
